std/algorithm/comparison.d

```diff
@@ -1964,60 +1950,31 @@ Returns:
     `true` if both ranges have the same length, `false` otherwise.
 */
 bool isSameLength(Range1, Range2)(Range1 r1, Range2 r2)
-if (isInputRange!Range1 &&
-    isInputRange!Range2 &&
-    !isInfinite!Range1 &&
-    !isInfinite!Range2)
+if (isInputRange!Range1 && isInputRange!Range2)
 {
-    static if (hasLength!(Range1) && hasLength!(Range2))
+    static if (isInfinite!Range1 || isInfinite!Range2)
+    {
+        return isInfinite!Range1 && isInfinite!Range2;
```

Comment:
Cantor disagrees:
- https://en.wikipedia.org/wiki/Cantor%27s_diagonal_argument
- https://en.wikipedia.org/wiki/Cantor%27s_theorem

(tl;dr: infinite sets can have different length)

Reply:
Heh, crossed my mind. Both are countable so they actually do have "the same" length.